engine/swarm/secrets.md

```diff
@@ -0,0 +1,771 @@
+this example. Also, changing a MySQL root passphrase isn’t as simple as changing
+a file on disk. You must use a query or a `mysqladmin` command to change the
+password in MySQL.
+
+1.  Generate a random alphanumeric password for MySQL and store it as a Docker
+    secret with the name `mysql_password` using the `docker secret create`
+    command. To make the password shorter or longer, adjust the last argument of
+    the `openssl` command. This is just one way to create a relatively random
+    password. You can use another command to generate the password if you
+    choose.
+
+    **Note**: After you create a secret, you cannot update it. You can only
+    remove and re-create it, and you cannot remove a secret that a service is
+    using. However, you can grant or revoke a running service's access to
+    secrets using `docker service update`. If you need the ability to update a
+    secret, consider adding a version component to the secret name, so that you
+    can later add a new version, update the service to use it, then remove the
+    old version.
+
+    ```bash
+    $ mkdir tempdir && openssl rand -base64 20 > tempdir/mysql_password.txt
+
+    $ docker secret create -f tempdir/mysql_password.txt mysql_password
+
+    lwxgks2rojdbvaui5kay1suvr
+    ```
+
+    The value returned is not the password, but the ID (digest) of the secret.
+    List the secrets managed by Docker using docker secret ls:
+
+    ```bash
+    $ docker secret ls
+
+    ID                          NAME                CREATED             UPDATED             SIZE
+    lwxgks2rojdbvaui5kay1suvr   mysql_password      41 seconds ago      41 seconds ago      29 B
+    ```
+
+    The secret is stored in the encrypted WAL logs for the swarm.
+
+2.  Create the MySQL container.  The MySQL container will have the following
+    characteristics:
+
+    - Exposes the port publicly because otherwise the WordPress service will not
+      be able to access MySQL. Non-service containers created with `docker run`
+      cannot join swarm overlay networks by default.  
+    - Persists MySQL data in a named volume (called `tempdir`) so that data
+      isn’t lost across restarts.  
+    - The `tempdir` volume containing the `mysql_password.txt` secret will be
+      mounted as a volume in the container so that the password is never exposed
+      as an environment variable, nor can it be committed to an image if the
+      `docker commit` command is run.
+    - Sets the environment variable `MYSQL_ROOT_PASSWORD_FILE` to point to the
+      `mysql` password in the bind-mounted `tempdir` volume. The `mysql` image
+      reads the password string from that file when initializing the database
+      for the first time. Afterward, it is stored in the MySQL database itself.
+
+      ```bash
+      $ docker run --name mysql -d \
+                   -v $(pwd)/tempdir:/initpw \
+                   -e MYSQL_ROOT_PASSWORD_FILE=/initpw/mysql_password.txt \
+                   -p 3306:3306 \
```

Comment:
I originally ran into errors attaching the container to the network, but looking it up some more it looks like I just needed to have an `--attachable` flag when creating the network?